app/controllers/results_controller.rb

```diff
@@ -4,12 +4,23 @@ class ResultsController < ApplicationController
 
 
   def create
-
     multiplier = params[:multiplier] || 1
+    opponent = params["result"]["teams"]["1"]["players"]
+    render :new && return if opponent.nil?
+
+    current_player = [@current_player.id.to_s]
+    winner = params["relation"] == "defeated" ? current_player : opponent
```

Comment:
this line and the one below is using `current_player` instead of reading from `params["result"]["teams"]["0"]` , so it's safe even people want to do something tricky and manually set `params["result"]["teams"]["0"]`.